RCS Refactor and Balancing

Fox-McCloud opened the pull request:
Updates, fixes, and tweaks the RCS (Rapid Crate Sender)

- Can now properly use the RCS on locked crates and lockers
- Now required to stand next to a crate/locker to teleport it until the teleport goes through.
 - it previously would just blindly teleport the locker 5 seconds after you initially selected its destination, regardless of adjacency.
- Gives the RCS the RCD in-hand icon
- Updates the description and emagged feature descriptions to sound a bit more IC.
- Can no longer teleport yourself around using the RCS
 - You can still teleport OTHER people using the RCS, but you can't hop in a closet and tele yourself.
- RCS now has a battery in it as opposed to relying on magical "rcs uses"; the unit will not self-recharge any longer, but will be able to be recharged in a standard charger.
- SPANSSSS

Map Related

- Removed the RCS from the mining asteroid station and moved it to where miners spawn
- Put a recharger where the RCS in minining used to be
- Put a charger in cargo office (miners and cargo techs can access this one)

Other

- Removes a duplicate instance of the telepad spawner in the code

Should better help the RCS be utilized as an actual cargo transportation device rather than a "herp derp, ghetto hand tele that never fails!"

Earthdivine commented:
>Can't teleport self

No fun allowed ;w;

Spacemanspark commented:
Eh.

tigercat2000 commented:
Really don't like removing the RCS teleporting yourself.

tigercat2000 commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -177,6 +177,8 @@
 
 /obj/structure/closet/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
 	if(istype(W, /obj/item/weapon/rcs) && !src.opened)
+		if(locate(user) in src.contents) //to prevent self-teleporting.
```
.. this is the most terrifying way you could possibly do this. 
If anything, it should look more like
```
if(user in contents) return
```

Dumbdumn5 commented:
I agree with tiger, teleporting yourself is a bit... yeah... you should probably be allowed to do that still. If the locker opens at all it gets teleported out without you inside, not a huge issue that I've seen involving the teleporting, especially for an item so rarely used except for mining and emergencies. Other than that, usage is incredibly rare, and now with the removal of Tsci, this is one of the few functioning teleporter systems aside from the larger teleporters out on the derelict regions and below the cap's office. 

I can understand the requirement of standing next to them, but why not have it act as pulling or a grab where it auto-cancels if you move away?

Markolie commented:
Perhaps make the pad upgradeable? And if it's fully upgraded, there's no delay and you can teleport yourself?

I don't really mind either way though.

Fox-McCloud commented:
Updated the RCS a bit more; it starts off with a cell in it now and pulls power from the cell when you use it as opposed to relying on magical "rcs uses". This means it will no longer self-recharge, however; you will, however, be able to recharge it in a recharge unit.

Fox-McCloud commented:
Moved the mining RCS and added a a couple of chargers so they can be recharged.

This PR ended up being way more entailed than I originally intended.

Spacemanspark commented:
I'd prefer we still be able to teleport ourselves.

Robertha89 commented:
Agreed, teleporting inside was what really made it an good item, since you could go to cargo, melt the stuff, grab it, go to science, upgrade your stuff, and go back to mining the entire asteroid..

Fox-McCloud commented:
That's really precisely my problem with it, @Robertha89; people aren't using or valuing it for the ability to transport crates, goods, etc to far off locations; they're valuing it for a *personal* self transportation device that has no consequences to use (unlike the hand tele and wormhole jaunter).

When I see players utilizing this to send people to unreachable spots that can't be escaped from, then mass converting said people and utilizing this device to repeatedly escape sec and NOT to transport crates? That tells me there's an inherent problem with the item's usage vs purpose.

Compare this to the hand tele, which there are only two of, it has a decent chance to fail (and space you), only the captain+2 heads of staff have access to, and you can be permabrigged for stealing it---vs an item that literally carries no ramifications for stealing other than a 10 minute sentence and is available to some of the lowest ranked personnel on the station.

Dumbdumn5 commented:
@Fox-McCloud There's only one RCS, yes there's little to no sentence, but it can only go to three set locations (Yes, I know you can move the pads) around the station, and good luck trying to get anybody to build more. As for sending people to unreachable spots randomly or even at all, I've never once seen a single case of it. I'm not entirely sure it's all too common, and I've only seen it used once by a cultist, and even then, there wasn't any massive conversion with it. As for escaping sec, I've really only seen that maybe three times. It could be that I'm not looking hard enough, but every time I observe the RCS is usually right on the table throughout the entire round, except ofr the one in mining, and that's usually used by the miners to transport minerals or indeed themselves, so that they can do their job more efficiently. If the teleporting people away would be such an issue, what about if you can just weld a locker full of people shut, RCS them off to a hidden maint area, and then mass convert them, or have a mining buddy teleport you off to science while you have the other RCS with you so you can hand it off to, I dunno, science, and have them teleport you back with your new loot.

Transporting crates quickly is nice, but I see the mailing system used more often than the RCS or hand delivering the crates. Sure it's got the value of personal transportation, but I really don't see how it's a big issue. If stealing it is such a big problem, shove it into the QM's locker instead of just sitting on a table in disposals.

Robertha89 commented:
if the handteleporter fails, you use it again, and portal again, and bam, you got 10 brute damage, 3 burn damage and 8 oxygen damage, instead of removing it, add the chance of failure, and with how it's  not auto-recharge, there will be a chance of permanent spacing. The issue with transporting crates and goods is that You have to trust others to do their job, and i've been a miner, teleported back half a dozen closets, gone to the station and see them all stacked upon each other, even then, Science have no RCS to send stuff to us with, so we need to go there anyway. I have also seen the handteleporter used to send people to unreachable spots that can't be escaped from, and then mass converting or mass murdering people, and even to escape sec, cause thats the points, the thing with the RCS is you need a closet, and you need the closet to be closed for 5 seconds, with teleporter, you can just teleport till you hit space, then they can't chase you no more, cause you can teleport to a new location then.

Also, also, there are only two rcs, which only what 8 people have access too, and only 3 of them ever takes it, being the miners, and only the miners has access to both. 

Also, do remember where the teleporter pads are, you need your stuff for RnD, not Telescience.

If it's being used to escape sec, good, it's a tool and should be used like that, it's even better to escape with if you have a bag of holding. Even then, it's not even commonly used like that, it's rare, like really rare, and is basically only used by miners to move from mining to rnd and back.

TheDZD commented:
The RCS is supposed to be used for teleporting crates, not as a ghetto (not even that ghetto considering it is fully accurate and has no ill effects) hand teleporter. Given that you can move telepads and even place new ones, it's more or less a better version of the stationary teleporter (no chance to accidentally space yourself), with the caveat that you need to shove yourself in a locker. Personal teleportation items should stay restricted to command, and your average miner shouldn't have an item that is as good as the teleporter room which is restricted to select heads of staff.

Dumbdumn5 commented:
Why not just make it so you can only teleport crate while they're on top of the actual telepads if they're so OP?

Fox-McCloud commented:
@Dumbdumn5 

Because it's a cargo transportation device; it's meant to make cargo techs and miner's lives easy and convenient for transporting and shipping deliveries---not a personal transportation device.

Robertha89 commented:
Then make it have a chance of failure instead of removing it as a tool, the station is big enough where having public teleporters could be a thing..

Fox-McCloud commented:
@Robertha89 

Then no one will use it, ever---because no one will want to their cargo points randomly eaten---but someone who's in a locker who gets spaced can just as easily use it again to return to the station.

That would make it *even more* of a personal transportation device than it already is.

Robertha89 commented:
Then it will never used, and can just be deleted from the code right now.

If you can't teleport back to station, then you can't get the cargo points, so go back yourself, and if your there why even bother teleporting it, just mail it. The fact that you can teleport with you inside is the only reason it's actually used, remove it and you can just remove the item because it will never be used.

Spacemanspark commented:
Please, just let us keep the damn ability.

Earthdivine commented:
While I was joking about the teleport removal, the RCS is literally useless regardless. It only works between mining and cargo (You need to be there to collect points now) and Science (who should be coming to get the minerals themselves.) The Mail System is just so superior, particullarly since the RCS system wasn't finished, as I'm sure it was meant to replace the Mules and Mail system.

I can hardly see anyone using this anymore, but having it being a Ghetto hand teleporter for free is certainly not the way to keep it alive. The fact people are complaining they can't teleport THEMSELVES when teleporting other people is still in proves it's a problem.
